### PR TITLE
render: reduce cpu consumption while idle

### PR DESCRIPTION
**Pull request by VioletRed**
Enforce max 100fps in GUI rendering.

### CODE REVIEW

File: xbmc/Application.cpp [hunk 1]
```diff
@@ -1990,17 +1990,17 @@ void CApplication::Render()
...
-    if (frameTime < singleFrameTime)
-      Sleep(singleFrameTime - frameTime);
+    singleFrameTime = 40; //if not flipping, loop at 25 fps
   }
 
+  //fps limiter, make sure each frame lasts at least singleFrameTime milliseconds
+  unsigned int frameTime = now - m_lastFrameTime;
+  if (frameTime < singleFrameTime)
+    Sleep(singleFrameTime - frameTime);
+
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: this seems wrong. we don't want this pulled out of the above block. does your system goes above 100fps?

Comment (MartijnKaijser) [inline xbmc/Application.cpp]: as note, i saw my ADT-1 going to 140fps in system info. So it does happen

Comment (VioletRed) [inline xbmc/Application.cpp]: I know it looks weird, but yes, it does happen that it goes too high if you have looping animations.

Also looking at line 1894
```
unsigned int singleFrameTime = 10; // default limit 100 fps
```
It looks like the limit was intended by design, just wrongly implemented.

Comment (FernetMenta) [inline xbmc/Application.cpp]: I don't see it implemented incorrectly. the limiter is supposed to only engage if vsync is off.

Comment (VioletRed) [inline xbmc/Application.cpp]: "Aha" moment of the day. Then the problem is with x11 windowing code. I'll take a look and come back.

Comment (FernetMenta) [inline xbmc/Application.cpp]: Are you on NVidia? I investigated an issue some time ago where vsync stopped working every now and then on NVidia hw but I was not able to reproduce on my systems.

**Comment by VioletRed** [comment]
I still can't find the problem, but will cancel the pull request anyway until I find more info.

**Comment by FernetMenta** [comment]
for reference https://github.com/FernetMenta/xbmc/issues/109

**Pull request closed without merging**